sc-memory/sc-memory/tests/sc-memory/units/templates/test_sc_templates_scs.cpp

```diff
@@ -21,13 +21,20 @@ TEST_F(ScTemplateSCsTest, BuildSuccessful)
   m_ctx->BuildTemplate(templ, data);
 }
 
-TEST_F(ScTemplateSCsTest, BuildFail)
+TEST_F(ScTemplateSCsTest, InvalidSCs)
 {
   ScTemplate templ;
   sc_char const * data = "_a _-> b";
   EXPECT_THROW(m_ctx->BuildTemplate(templ, data), utils::ExceptionParseError);
 }
 
+TEST_F(ScTemplateSCsTest, ElementNotFound)
+{
+  ScTemplate templ;
+  sc_char const * data = "_a _-> b;;";
+  EXPECT_THROW(m_ctx->BuildTemplate(templ, data), utils::ExceptionInvalidState);
```

Comment:
why is it throwing?

Reply:
sc-element with system identifier `b` doesn't exist. BuildTemplate doesn't generate anything in sc-memory